packages/platforms/browser/lib/backgrounding-listener.ts

```diff
@@ -1,35 +1,61 @@
 import {
   type BackgroundingListener,
-  type BackgroundingListenerCallback
+  type BackgroundingListenerCallback,
+  type BackgroundingListenerState
 } from '@bugsnag/core-performance'
 
+interface EventListenerOptions {
+  once?: boolean
+}
+
 interface DocumentForVisibilityState {
   addEventListener: (event: string, callback: () => void) => void
   visibilityState: string
 }
 
-export default function createBrowserBackgroundingListener (document: DocumentForVisibilityState) {
+interface WindowWithDocumentForVisibilityState {
+  document: DocumentForVisibilityState
+  addEventListener: (event: string, callback: () => void, options?: EventListenerOptions) => void
+}
+
+export default function createBrowserBackgroundingListener (window: WindowWithDocumentForVisibilityState) {
   const callbacks: BackgroundingListenerCallback[] = []
 
   const backgroundingListener: BackgroundingListener = {
     onStateChange (backgroundingListenerCallback: BackgroundingListenerCallback): void {
       callbacks.push(backgroundingListenerCallback)
 
       // trigger the callback immediately if the document is already 'hidden'
-      if (document.visibilityState === 'hidden') {
+      if (window.document.visibilityState === 'hidden') {
         backgroundingListenerCallback('in-background')
       }
     }
   }
 
-  document.addEventListener('visibilitychange', function () {
-    const state = document.visibilityState === 'hidden'
-      ? 'in-background'
-      : 'in-foreground'
-
+  const notifySubscribers = (state: BackgroundingListenerState) => {
     for (const callback of callbacks) {
       callback(state)
     }
+  }
+
+  window.document.addEventListener('visibilitychange', function () {
+    const state = window.document.visibilityState === 'hidden'
+      ? 'in-background'
+      : 'in-foreground'
+
+    notifySubscribers(state)
```

Comment:
Would it be sensible to keep track of the last known state and only notify subscribers when it changes?  To avoid getting multiple notifications for effectively the same event?

Reply:
Yes it would!